File: java/src/privileges/Publish.java
package privileges;

import decorator.PrivilegeDecorator;
import decorator.User;

public class Publish extends PrivilegeDecorator{
	User user;
	
	public Publish(User user) {
		this.user = user;
	}

	@Override
	public String getPrivileges() {
		return user.getPrivileges() + " publish";
	}

	@Override
	public String getName() {
		return user.getName();
	}
}

File: csharp/decorator_practice/decorator_practice/main/privileges/Publish.cs
using decorator_practice.main.decorator;
using System;
using System.Collections.Generic;
using System.Text;

namespace decorator_practice.main.privileges
{
    public class Publish : PrivilegeDecorator
    {
        /**
	     * User.
	     */
        private User user;

        /**
	     * @param u constructor.
	     */
        public Publish(User u)
        {
            this.user = u;
        }

        /**
	     * @return approve privilege.
	     */
        override
        public String getPrivileges()
        {
            return user.getPrivileges() + " publish";
        }

        /**
	     * @return name.
	     */
        override
        public String getName()
        {
            return user.getName();
        }
    }
}
